### PR TITLE
feat: query theme support

### PR DESCRIPTION
**Pull request by manneredboor**
### Description

Theme adjustment from query param `?theme=light` or `?theme=dark`

### Checklist:

- [x] Checked the changes locally.

### CODE REVIEW

**Review by DiRaiks — DISMISSED** [review]

**Review by Jeday — DISMISSED** [review]

File: providers/theme.tsx [hunk 1]
```diff
@@ -1,103 +1,21 @@
...
-import { BackgroundGradient } from 'shared/components';
+import { FC, PropsWithChildren, useMemo } from 'react';
+import { CookieThemeProvider, ThemeName } from '@lidofinance/lido-ui';
+import { useRouter } from 'next/router';
+export { MODAL, ModalContext } from './modals';
 
-export type ThemeName = 'light' | 'dark';
+export const ThemeProvider: FC<PropsWithChildren> = ({ children }) => {
+  const router = useRouter();
+  const queryTheme = router.query.theme;
```
Comment (solidovic) [inline providers/theme.tsx:8]: may be should add optional chaining operator?

```
const queryTheme = router?.query?.theme;
```

File: providers/theme.tsx [hunk 2]
```diff
@@ -1,103 +1,21 @@
...
-
-  const value = useMemo(
-    () => ({
-      toggleTheme,
-      themeName,
-    }),
-    [themeName, toggleTheme],
-  );
+  const overrideThemeName: ThemeName | undefined = useMemo(() => {
+    if (queryTheme === 'light') return ThemeName.light;
```
Comment (solidovic) [inline providers/theme.tsx:11]: a border case but I suggest take into account the case `queryTheme = Light` or another variants

```queryTheme?.toLowerCase() ```

File: providers/theme.tsx (hunk 1 above)
Comment (manneredboor) [reply providers/theme.tsx:8]: We could do it certainly but it does not seems to be necessary according next.js typings

Comment (manneredboor) [reply providers/theme.tsx]: Done it anyway

**Review by solidovic — DISMISSED** [review]
Lgtm

**Review by DiRaiks — APPROVED** [review]

**Review by DiRaiks — CHANGES_REQUESTED** [review]
/rewards page crashes after reload

**Review by Jeday — APPROVED** [review]

**Review by DiRaiks — APPROVED** [review]

**Pull request merged**